hw/syn/data/common_syn_cfg.hjson

```diff
@@ -0,0 +1,28 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  project:          opentitan
+  doc_server:       docs.opentitan.org
+  results_server:   reports.opentitan.org
+
+  flow:             syn
+  flow_makefile:    "{proj_root}/hw/syn/data/syn.mk"
+
+  import_cfgs:      [// common server configuration for results upload
+                     // "{proj_root}/hw/dv/data/fusesoc.hjson",
+                     "{proj_root}/hw/data/common_project_cfg.hjson",
+                     "{proj_root}/hw/syn/data/{tool}.hjson"]
+
+  // Default directory structure for the output
+  dut:              "{name}"
+  build_dir:        "{scratch_path}/{build_mode}"
+
+  // We rely on Fusesoc to generate the file list for us
+  sv_flist_gen_cmd:   fusesoc
+  fusesoc_core_:      "{eval_cmd} echo \"{fusesoc_core}\" | tr ':' '_'"
+  sv_flist_gen_opts:  ["--cores-root {proj_root} ",
+                       "run --target=syn --tool icarus --build-root={build_dir}",
```

Comment:
If this is only creating the filelist, then we should have been able to reuse `hw/dv/data/fusesoc.hjson`. Does the --target=syn switch do anything special?

Reply:
Please file an issue in OT and add a TODO comment to switch this to the real DC edalize backend once we have that.

Reply:
yes the `--target=syn` switch causes fusesoc to use a different target in the core file, where you may do things like specifying additional tool parameters or Verilog DEFINES. tool parameters are currently not really possible yet since, as @imphil mentioned, we do not yet have the right edalize backend for DC in place. but defines can already be used. they get dumped into the file list, and then they are used during analysis and elaboration in DC.

Reply:
Added a TODO and another bullet point here #1727. Edalize already has an associated issue:
https://github.com/olofk/edalize/issues/89
I am going to update the script example that I posted on that issue in a minute.